### PR TITLE
added notification for dateTime

### PR DESCRIPTION
**Pull request by fbergunde**

### CODE REVIEW

File: src/rumo-datetime-picker.html [hunk 1]
```diff
@@ -168,14 +177,29 @@
        * @return {Date} The calculated date.
        */
       _computeDateTime(date, time) {
+        if (this._noComputation) {
+          this._noComputation = false;
+          return;
+        }
```
Comment (stefanfrede) [inline src/rumo-datetime-picker.html]: What is the goal of a `_noComputation`-property in a computed properties method?

File: src/rumo-datetime-picker.html [hunk 2]
```diff
@@ -112,7 +112,9 @@
           /** The currently selected date and time as date object  */
           dateTime: {
             type: Object,
-            computed: '_computeDateTime(_date, _time)',
+            // computed: '_computeDateTime(_date, _time)',
+            notify: true,
+            observer: '_dateTimeUpdate',
```
Comment (stefanfrede) [inline src/rumo-datetime-picker.html]: Why remove the computed property? It behaves like an observable and notifies about updates, so there is no need to implement it yourself. Have a look here: https://www.polymer-project.org/2.0/docs/devguide/observers#computed-properties.

File: src/rumo-datetime-picker.html [hunk 3]
```diff
@@ -144,6 +146,13 @@
         Sugar.extend();
       }
 
+      // (Complex) Observers
+      static get observers() {
+        return [
+          '_computeDateTime(_date, _time)'
+        ]
+      }
+
```
Comment (stefanfrede) [inline src/rumo-datetime-picker.html]: Unnecessary if you use the computed properties method.

File: src/rumo-datetime-picker.html [hunk 4]
```diff
@@ -168,14 +177,29 @@
...
-          return dateTime;
+      /* listen to changes of dateTime */
+      _dateTimeUpdate(dateTime) {
+        if (this._noUpdate) {
+          this._noUpdate = false;
+          return;
         }
+        this._noComputation = true;
+        this.set("_date", Sugar.Date.format(this.dateTime, '%Y-%m-%d'));
+        this.set("_time", Sugar.Date.format(this.dateTime, '%H:%M'));
```
Comment (stefanfrede) [inline src/rumo-datetime-picker.html]: It looks to me that you are manually adding the functionality you get with using a computed property. Why?

File: src/rumo-datetime-picker.html (hunk 2 above)
Comment (stefanfrede) [reply src/rumo-datetime-picker.html]: Ok. Got it.

File: src/rumo-datetime-picker.html [hunk 5]
```diff
@@ -112,8 +112,9 @@
           /** The currently selected date and time as date object  */
           dateTime: {
             type: Object,
-            computed: '_computeDateTime(_date, _time)',
+            // computed: '_computeDateTime(_date, _time)',
```
Comment (stefanfrede) [inline src/rumo-datetime-picker.html]: Please remove the comment.

File: src/rumo-datetime-picker.html [hunk 6]
```diff
@@ -179,6 +187,12 @@
         }
       }
 
+      /* listen to changes of dateTime */
```
Comment (stefanfrede) [inline src/rumo-datetime-picker.html]: Please stick to the guidelines when documenting methods: https://www.polymer-project.org/2.0/docs/tools/documentation#methods

**Comment by stefanfrede** [comment]
Please don't forget to run `npm version minor` to update the version (https://github.com/arigo-software/frontend-development-guidelines => Versioning).

File: src/rumo-datetime-picker.html (hunk 5 above)
Comment (fbergunde) [reply src/rumo-datetime-picker.html]: Done

File: src/rumo-datetime-picker.html (hunk 3 above)
Comment (fbergunde) [reply src/rumo-datetime-picker.html]: I'll just remove the comment.

File: src/rumo-datetime-picker.html (hunk 4 above)
Comment (fbergunde) [reply src/rumo-datetime-picker.html]: Discussed in a zoom-conversation.

File: src/rumo-datetime-picker.html (hunk 1 above)
Comment (fbergunde) [reply src/rumo-datetime-picker.html]: Discussed in a zoom-conversation.

File: src/rumo-datetime-picker.html (hunk 3 above)
Comment (fbergunde) [reply src/rumo-datetime-picker.html]: Discussed in a zoom-conversation.

File: src/rumo-datetime-picker.html (hunk 4 above)
Comment (fbergunde) [reply src/rumo-datetime-picker.html]: Discussed in a zoom-conversation.

File: src/rumo-datetime-picker.html (hunk 1 above)
Comment (fbergunde) [reply src/rumo-datetime-picker.html]: Discussed in a zoom-conversation.

File: src/rumo-datetime-picker.html (hunk 5 above)
Comment (fbergunde) [reply src/rumo-datetime-picker.html]: Done.

File: src/rumo-datetime-picker.html [hunk 7]
```diff
@@ -178,26 +176,26 @@
...
       _dateTimeUpdate(dateTime) {
         if (this._noUpdate) {
-          this._noUpdate = false;
+          this.set("_noUpdate", false);
           return;
         }
-        this._noComputation = true;
+        this.set("_noComputation", true);
         this.set("_date", Sugar.Date.format(this.dateTime, '%Y-%m-%d'));
         this.set("_time", Sugar.Date.format(this.dateTime, '%H:%M'));
```
Comment (stefanfrede) [inline src/rumo-datetime-picker.html]: A couple of things are missing:

+ Please add `_noComputation` to the properties list (`_noComputation: Boolean,`).
+ Please add `_noUpdate` to the properties list (`_noUpdate: Boolean,`).
+ Please provide a summary for the `_dateTimeUpdate`-method.

```
/**
 * Short description …
 *
 * @param {Object} dateTime Explanation …
 */
```

Relate to https://www.polymer-project.org/3.0/docs/tools/documentation#type-annotation and http://usejsdoc.org/.

+ Please run `polymer analyze > analysis.json` to update the documentation.
+ Please add yourself to the contributors array in the `package.json`.
+ Please solve the conflicts.

Once we merged into the `master` I will take care of the accidental version bump.

**Pull request merged**